File: VitalSphere/VitalSphere.Services/Database/WellnessBox.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace VitalSphere.Services.Database
{
    public class WellnessBox
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [MaxLength(150)]
        public string Name { get; set; } = string.Empty;

        [MaxLength(500)]
        public string? Description { get; set; }

        public byte[]? Image { get; set; }

        [MaxLength(1000)]
        public string? IncludedItems { get; set; }

        public bool IsActive { get; set; } = true;

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public ICollection<Gift> Gifts { get; set; } = new List<Gift>();
    }
}

File: VitalSphere/UI/vital_sphere_mobile/lib/model/wellness_box.dart
import 'package:json_annotation/json_annotation.dart';

part 'wellness_box.g.dart';

@JsonSerializable()
class WellnessBox {
  final int id;
  final String name;
  final String? description;
  final String? image;
  final String? includedItems;
  final bool isActive;
  final DateTime createdAt;

  WellnessBox({
    this.id = 0,
    this.name = '',
    this.description,
    this.image,
    this.includedItems,
    this.isActive = true,
    required this.createdAt,
  });

  factory WellnessBox.fromJson(Map<String, dynamic> json) =>
      _$WellnessBoxFromJson(json);
  Map<String, dynamic> toJson() => _$WellnessBoxToJson(this);
}

